### ISSUE TITLE
Deterministic Serialization?

### ISSUE THREAD

**Issue by AlgorithmsAreCool** [serdedotnet/serde#142]
Hey this is a great library!

I was wondering if there are any guarantees around deterministic serialization. Is it possible to annotate properties with an order attribute that will guarantee the serialization order of properties?

**Comment by agocke** [comment]
It should be as deterministic as the Roslyn APIs... which I believe are deterministic and return properties in alphabetical order. Obviously, if you add or remove properties that guarantee goes out the window. But at that point you could choose to implement `ISerialize` yourself and you would get the exact same code every time, and it would only change when you change it.

**Comment by AlgorithmsAreCool** [comment]
> But at that point you could choose to implement ISerialize yourself and you would get the exact same code every time, and it would only change when you change it.
 
Needing to provide an efficient bespoke implementation of `ISerialize` weakens the ergonomics of the library quite a bit.

That being said, now that I've played with the library some, I've realized that I can handle that requirement entirely inside of the `ISerializer` implementation (via buffering and sorting of the dictionary elements and type members).

There might be some utility in the future of allowing explicit control of member ordering via an attribute. But because I don't have a demonstrable need for this at the moment, I'm closing this.
